sway-core/src/semantic_analysis/ast_node/declaration/impl_trait.rs

```diff
@@ -663,12 +665,26 @@ fn type_check_trait_implementation(
                 let decl_ref = decl_engine.insert(type_decl.clone());
                 impld_item_refs.insert((name, implementing_for), TyTraitItem::Type(decl_ref));
 
-                let old_type_decl_info = TypeInfo::TraitType {
+                let old_type_decl_info1 = TypeInfo::TraitType {
                     name: type_decl.name.clone(),
                     trait_type_id: implementing_for,
                 };
+                let old_type_decl_info2 = TypeInfo::TraitType {
+                    name: type_decl.name.clone(),
+                    trait_type_id: type_engine.insert(
+                        engines,
+                        TypeInfo::UnknownGeneric {
+                            name: Ident::new_with_override("Self".into(), Span::dummy()),
```

Comment:
It would be great if anytime a `Span::dummy()` was used, a comment was added above it to explain why this token doesn't have a valid span and if it's expected to resolved to one later. If not, it's going to cause a bug in LSP.